main.go

```diff
@@ -37,40 +41,85 @@ func main() {
 		logrus.WithError(err).Fatal()
 	}
 
+	m := melody.New()
+
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
-	handleInterruptions(cancel)
+	handleInterruptions(cancel, func() error {
+		return m.CloseWithMsg(melody.FormatCloseMessage(1000, "The server says goodbye :)"))
+	})
 
-	// At this moment, we do nothing else.
-	go logValidationErrors(ctx, validationErrChan)
+	r := chi.NewRouter()
+	r.Get("/", http.FileServer(http.Dir("./web/static")).ServeHTTP)
```

Comment:
All this logic could be eventually moved to `proxy` package (`server.go` file for example), but for simplicity now I prefer to keep it here.